src/Command/PelagosMigrateDatasetFilesCommand.php

```diff
@@ -0,0 +1,128 @@
+    {
+        $this->entityManager = $entityManager;
+        // It is required to call parent constructor if
+        // using a constructon in a Symfony command.
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('This command will migrate the datastore to the new datastore.')
+            ->addOption('qf', null, InputOption::VALUE_NONE, 'Queue Dataset to Filer')
+            ->addOption('ifc', null, InputOption::VALUE_NONE, 'Do not check if dataset file exist on disk.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $io->title('Migrating Datasets Files');
+        
+        if ($input->getOption('qf')) {
+            $io->text('Also queueing the filer!');
+            $queueFiler = true;
+        } else {
+            $queueFiler = false;
+        }
+        
+        $datasetRepository = $this->entityManager->getRepository(Dataset::class);
+        
+        $datasets = $datasetRepository->findAll();
+        
+        foreach ($datasets as $dataset) {
+            $udi = $dataset->getUdi();
+
+            $status = $dataset->getDatasetStatus();
+            
+            $io->note(sprintf('UDI = %s.', $udi));
+            
+            if ($status === Dataset::DATASET_STATUS_IN_REVIEW) {
+                $datasetSubmission = $dataset->getLatestDatasetReview();
+                $this->setFile($datasetSubmission);
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                $this->setFile($datasetSubmission);
+            } else {
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                $this->setFile($datasetSubmission);
+            }
+            
+            $this->entityManager->persist($dataset);
+            
+        }
+
+        $this->entityManager->flush();
+        $io->success('Done!');
+
+        return 0;
+    }
+    
+    function setFile($datasetSubmission)
+    {
+        
+        if ($datasetSubmission instanceof DatasetSubmission) {
+            $udi = $datasetSubmission->getDataset()->getUdi();
+            
+            // TODO: Run querybuild to get filename and hash and size...
+            
+            // Todo: optional NOT file exist.
+            
+            $file = new File();
+            $file->setFilePathName($datasetSubmission->getDatasetFileName());
+            $file->setFileSize($datasetSubmission->getDatasetFileSize());
+            $file->setFileSha256Hash($datasetSubmission->getDatasetFileSha256Hash());
+
+            $file->setUploadedAt($datasetSubmission->getModificationTimeStamp());
+            $file->setUploadedBy($datasetSubmission->getModifier());
+
+            $file->setDescription('Original Dataset File');
+
+            $file->setPhysicalFilePath("/san/data/store/$udi/$udi.dat");
```

Comment:
perhaps get from the DATA_STORE_DIRECTORY env variable

Reply:
It's a parameter now, you review this just as I pushed an update.